File: common_library/src/main/java/com/phone/common_library/base/BasePresenter.java
package com.phone.common_library.base;

import java.lang.ref.WeakReference;

/**
 * author    : Urasaki
 * e-mail    : 1164688204@qq.com
 * date      : 2019/3/7 10:55
 * introduce :
 */


public class BasePresenter<T> {

    private WeakReference<T> modelView;

    protected void attachView(T view) {
        modelView = new WeakReference<>(view);
    }

    protected T obtainView() {
        return isAttach() ? modelView.get() : null;
    }

    private boolean isAttach() {
        return modelView != null &&
                modelView.get() != null;
    }

    public void detachView() {
        if (isAttach()) {
            modelView.clear();
            modelView = null;
        }
    }

}

File: library_common/src/main/java/com/phone/common_library/base/BasePresenter.kt
package com.phone.common_library.base

import java.lang.ref.WeakReference

/**
 * author    : Urasaki
 * e-mail    : 1164688204@qq.com
 * date      : 2019/3/7 10:55
 * introduce :
 */

open class BasePresenter<T> {

    private var modelView: WeakReference<T?>? = null

    protected fun attachView(view: T) {
        modelView = WeakReference(view)
    }

    protected fun obtainView(): T? {
        return if (isAttach()) modelView?.get() else null
    }

    private fun isAttach(): Boolean {
        return modelView != null &&
                modelView?.get() != null
    }

    open fun detachView() {
        if (isAttach()) {
            modelView?.clear()
            modelView = null
        }
    }

}